src/citrine/informatics/constraints/integer_range_constraint.py

```diff
@@ -0,0 +1,48 @@
+from typing import Optional
+
+from citrine._serialization import properties
+from citrine._serialization.serializable import Serializable
+from citrine.informatics.constraints.constraint import Constraint
+
+__all__ = ['IntegerRangeConstraint']
+
+
+class IntegerRangeConstraint(Serializable['IntegerRangeConstraint'], Constraint):
+    """Represents an inequality constraint on an integer-valued material attribute.
+
+    Parameters
+    ----------
+    descriptor_key: str
+        the key corresponding to a descriptor
+    lower_bound: int
+        the minimum value in the range
+    upper_bound: int
+        the maximum value in the range
+    lower_inclusive: bool
+        if True, will include the lower bound value in the range (default: true)
+    upper_inclusive: bool
+        if True, will include the max value in the range (default: true)
```

Comment:
Do we need to expose inclusive options? After all, the integer constraint [a, b] with exclusive bounds can also be written as [a+1, b-1] with inclusive bounds

Reply:
Hmm, idk why I ever added that in the backend. I think it was already there in mithril. Should probably just remove it.

Reply:
Im just gonna remove from backend and here.